content/docs/getting-started/runtime.md

```diff
@@ -54,29 +54,25 @@ trait directly.
 We can spawn tasks using `tokio::spawn`. For example:
 
 ```rust
-# extern crate tokio;
-# extern crate futures;
-#
 # use tokio::prelude::*;
-# use futures::stream;
+
 # fn main() {
-# let my_outer_stream = stream::once(Ok(1));
+# let mut my_outer_stream = tokio::stream::iter(Some(1));
 // Create some kind of future that we want our runtime to execute
-let program = my_outer_stream.for_each(|my_outer_value| {
-  println!("Got value {:?} from the stream", my_outer_value);
-
-  let my_inner_future = future::ok(1);
+let program = async move {
+  while let Some(my_outer_value) = my_outer_stream.next().await {
+    println!("Got value {:?} from the stream", my_outer_value);
 
-  let task = my_inner_future.and_then(|my_inner_value| {
-    println!("Got a value {:?} from second future", my_inner_value);
-    Ok(())
-  });
+    let my_inner_future = tokio::future::ready(1);
 
-  tokio::spawn(task);
-  Ok(())
-});
+    tokio::spawn(async move {
+      let my_inner_value = my_inner_future.await;
+      println!("Got a value {:?} from second future", my_inner_value);
+    });
+  }
+};
 
-tokio::run(program);
+tokio::runtime::Runtime::new().unwrap().block_on(program);
```

Comment:
why not just use `#[tokio::main]` in the example?

Reply:
We would have to define `my_outer_stream` which I wanted to avoid.

Reply:
ah, I see, that makes sense. we could always show a function like
```rust
fn make_stream() -> impl Stream<Item = usize> + Send + Sync {
# tokio::future::ready(1)
    // ...
}
```
and then call that function in the example. although I am not sure if this is any better.